### PR TITLE
refactor(common/resources): web-environment package

### PR DESCRIPTION
**Pull request by jahorton**
Addresses #3152, but likely doesn't fully 'fix' it.  Also note [this comment](https://github.com/keymanapp/keyman/pull/3146#issuecomment-632939655) referenced by the issue.

While this PR doesn't add a baseline tsconfig.json for use across the packages, it does help to clean up the `environment.inc.ts` pollution across the repo, centralizing it within a single build that occurs as part of our repo's package bootstrapping process.

We can always address the baseline `tsconfig.json` at a later point; we'll probably need some discussion regarding what should actually go into it, since our TS projects are actually fairly diverse in their needs.

---

Note that while I _could_ have the resulting `@keymanapp/web-environment` completely compiled into `@keymanapp/web-utils`'s code via `<reference path/>`, that would rapidly break if/when we start converting the TS code to modules.  As a result, I opted for the current approach.

### CODE REVIEW

File: resources/web-environment/.gitignore [hunk 1]
```diff
@@ -0,0 +1,2 @@
+environment.inc.ts
```
Comment (darcywong00) [inline resources/web-environment/.gitignore:1]: Should this be cleaned up (removed) in the other common/core/web/**/.gitignore files?

Comment (jahorton) [reply resources/web-environment/.gitignore:1]: Eventually, but that would require each of us to do a repo-wide cleaning of this file on our development machines.  I didn't want to enforce that quite yet, as I've come rather close to accidentally committing `.gitignore`'d files when branch-hopping and would rather not 'chance' that happening.

**Comment by darcywong00** [comment]
Does /developer/js/package.json need this change too?

**Comment by jahorton** [comment]
> Does /developer/js/package.json and tsconfig.json need this change too?

It does not appear so - there's no use of `@keymanapp/web-utils`, which is what propagated the requirement to the other packages.  Note that `models-templates` and `models-wordbreakers` also do not use the new package.

**Pull request merged**